test/integration/connection-test.coffee

```diff
@@ -1020,3 +1020,68 @@ exports.disableAnsiNullDefault = (test) ->
     runSqlBatch test, config, sql, (err) ->
         test.ok(err instanceof Error)
         test.strictEqual err?.number, 515 # Cannot insert the value NULL
+
+
+# Test that the default behaviour has ARITHABORT set to off
+exports.testArithAbortDefault = (test) ->
+	test.expect(5)
+
+	config = getConfig()
+	request = new Request('SELECT SESSIONPROPERTY(\'ARITHABORT\') AS ArithAbortSetting', (err, rowCount) ->
+		test.ifError(err)
+		test.strictEqual(rowCount, 1)
+
+		connection.close()
+	)
+
+	request.on('columnMetadata', (columnsMetadata) ->
+		test.strictEqual(Object.keys(columnsMetadata).length, 1)
+	)
+
+	request.on('row', (columns) ->
+		test.strictEqual(Object.keys(columns).length, 1)
+		test.strictEqual(columns[0].value, 0)
+	)
+
+	connection = new Connection(config)
+
+	connection.on('connect', (err) ->
+		connection.execSql(request)
+	)
+
+	connection.on('end', (info) ->
+		test.done()
+	)
+
+# Test that ARITHABORT can be set to on
+exports.testArithAbortCanBeSetToOn = (test) ->
```

Comment:
These tests are almost identical, except one checks for '0' and the other for '1' for value of arithabort. Please refactor the common implementation. Something like:

function testArithAbortImpl(test, arithAbortSetting)
..... Implementation here. ....

exports.testArithAbortDefault = (test) -> 
    testArithAbortImpl(test, undefined)

exports.testArithAbortOff = (test) -> 
    testArithAbortImpl(test, false)

exports.testArithAbortOn = (test) -> 
    testArithAbortImpl(test, true)